frontend/src/components/core/Block/ElementNodeRenderer.tsx

```diff
@@ -0,0 +1,588 @@
+// AutoSizer causes it to rerender multiple times for different widths
+// when the sidebar is toggled, which significantly slows down the app.
+const BokehChart = React.lazy(() =>
+  import("src/components/elements/BokehChart/")
+)
+const DebouncedBokehChart = debounceRender(BokehChart, 100)
+
+const DataFrame = React.lazy(() =>
+  import("src/components/elements/DataFrame/")
+)
+const DeckGlJsonChart = React.lazy(() =>
+  import("src/components/elements/DeckGlJsonChart/")
+)
+const GraphVizChart = React.lazy(() =>
+  import("src/components/elements/GraphVizChart/")
+)
+const IFrame = React.lazy(() => import("src/components/elements/IFrame/"))
+const ImageList = React.lazy(() =>
+  import("src/components/elements/ImageList/")
+)
+const PlotlyChart = React.lazy(() =>
+  import("src/components/elements/PlotlyChart/")
+)
+const VegaLiteChart = React.lazy(() =>
+  import("src/components/elements/VegaLiteChart/")
+)
+const Video = React.lazy(() => import("src/components/elements/Video/"))
+
+// Lazy-load widgets.
+const Button = React.lazy(() => import("src/components/widgets/Button/"))
+const DownloadButton = React.lazy(() =>
+  import("src/components/widgets/DownloadButton/")
+)
+const Checkbox = React.lazy(() => import("src/components/widgets/Checkbox/"))
+const ColorPicker = React.lazy(() =>
+  import("src/components/widgets/ColorPicker")
+)
+const DateInput = React.lazy(() => import("src/components/widgets/DateInput/"))
+const Multiselect = React.lazy(() =>
+  import("src/components/widgets/Multiselect/")
+)
+const Progress = React.lazy(() => import("src/components/elements/Progress/"))
+const Radio = React.lazy(() => import("src/components/widgets/Radio/"))
+const Selectbox = React.lazy(() => import("src/components/widgets/Selectbox/"))
+const Slider = React.lazy(() => import("src/components/widgets/Slider/"))
+const FileUploader = React.lazy(() =>
+  import("src/components/widgets/FileUploader/")
+)
+const TextArea = React.lazy(() => import("src/components/widgets/TextArea/"))
+const TextInput = React.lazy(() => import("src/components/widgets/TextInput/"))
+const TimeInput = React.lazy(() => import("src/components/widgets/TimeInput/"))
+const NumberInput = React.lazy(() =>
+  import("src/components/widgets/NumberInput/")
+)
+
+interface ElementNodeRendererProps extends CommonProps {
+  node: ElementNode
+  width?: number
+}
+
+const RawElementNodeRenderer = (
+  props: ElementNodeRendererProps
+): ReactElement => {
+  const { node } = props
+
+  if (!node) {
+    throw new Error("ElementNode not found.")
+  }
+
+  const widgetProps = {
+    widgetMgr: props.widgetMgr,
+    disabled: props.widgetsDisabled,
+  }
+
+  let height: number | undefined
+
+  // TODO: Move this into type signature of props. The width is actually guaranteed to be nonzero
+  // since leaf elements are always direct children of a VerticalBlock, which always calculates
+  let width = props.width ?? 0
```

Comment:
Is this a TODO because following through on it would require a lot more work?

Reply:
Not a lot more work, but enough work that the PR would become a little more annoying to read. Anyway, I just went ahead and made the change since you seem to be in a PR reviewing mood 😉